### PR TITLE
StructureDefinition-cqfRecorded

### PR DESCRIPTION
**Pull request by bryantaustin13**
https://jira.hl7.org/browse/FHIR-45369 Update QI-Core to use extensions pack
This adds the cqfRecorded StructureDefinition

### CODE REVIEW

File: input/definitions/StructureDefinition/StructureDefinition-cqf-recorded.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  "experimental": false,
+  "date": "2018-08-22",
+  "publisher" : "HL7 International / FHIR Infrastructure",
+  "contact": [
+    {
+      "name" : "Clinical Quality Information WG",
+      "telecom": [
+        {
+          "system": "url",
+          "value" : "http://www.hl7.org/Special/committees/fiwg"
```
Comment (brynrhodes) [inline input/definitions/StructureDefinition/StructureDefinition-cqf-recorded.json:21]:
```suggestion
          "value" : "http://www.hl7.org/Special/committees/cqi"
```

File: input/definitions/StructureDefinition/StructureDefinition-cqf-recorded.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      "uri": "http://hl7.org/v3",
+      "name": "RIM Mapping"
+    }
+  ],
+  "kind": "complex-type",
+  "abstract": false,
+  "context": [
+    {
+      "type": "element",
+      "expression": "Resource"
```
Comment (brynrhodes) [inline input/definitions/StructureDefinition/StructureDefinition-cqf-recorded.json:50]: This should be limited to only those resources that we expect this is usable on. In particular, it should not be usable on any resource that already has a "recorded date" element. In QICore, we only use this on Communication, MedicationAdministration, MedicationDispense, and Procedure.

**Review by brynrhodes — CHANGES_REQUESTED** [review]
Also, please move the definition to the definitions/Resource folder

File: input/definitions/StructureDefinition/StructureDefinition-cqf-recorded.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    "status": "extensions"
+  },
+  "url": "http://hl7.org/fhir/StructureDefinition/cqf-recorded",
+  "version": "5.0.0",
+  "name": "CQFNotDoneRecorded",
+  "title": "CQF Not Done Recorded",
+  "status": "active",
+  "experimental": false,
+  "date": "2018-08-22",
+  "publisher" : "HL7 International / FHIR Infrastructure",
```
Comment (brynrhodes) [inline input/definitions/StructureDefinition/StructureDefinition-cqf-recorded.json:14]:
```suggestion
  "publisher" : "HL7 International / Clinical Quality Information",
```

File: input/definitions/StructureDefinition/StructureDefinition-cqf-recorded.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      ]
+    }
+  ],
+  "description": "Captures the recorded date of the event.",
+  "jurisdiction": [
+    {
+      "coding": [
+        {
+          "system": "urn:iso:std:iso:3166",
+          "code": "US"
```
Comment (brynrhodes) [inline input/definitions/StructureDefinition/StructureDefinition-cqf-recorded.json:32]: Remove this jurisdiction, it's not appropriate in the extensions pack

**Comment by bryantaustin13** [comment]
Change requests addressed. Should be ready to be reviewed again.

**Review by brynrhodes — APPROVED** [review]

**Comment by brynrhodes** [comment]
Based on discussion with FHIR-I, this extension should be changed to be named event-recorded and the context should be changed to Communication only.

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
+{
+  "resourceType": "StructureDefinition",
+  "id": "cqf-event-recorded",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:3]:
```suggestion
  "id": "event-recorded",
```

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  "extension": [{
+    "url": "http://hl7.org/fhir/StructureDefinition/structuredefinition-wg",
+    "valueCode": "cqi"
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:6]:
```suggestion
    "valueCode": "fhir-i"
```

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  }],"text": {
+    "status": "extensions"
+  },
+  "url": "http://hl7.org/fhir/StructureDefinition/cqf-event-recorded",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:10]:
```suggestion
  "url": "http://hl7.org/fhir/StructureDefinition/event-recorded",
```

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  "version": "5.0.0",
+  "name": "CQFNotDoneRecorded",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:12]:
```suggestion
  "name": "EventRecorded",
```

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+  "title": "CQF Not Done Recorded",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:13]:
```suggestion
  "title": "Event Recorded",
```

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+  "status": "active",
+  "experimental": false,
+  "date": "2018-08-22",
+  "publisher" : "HL7 International / Clinical Quality Information",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:17]:
```suggestion
  "publisher" : "HL7 International / FHIR Infrastructure",
```

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+  "contact": [
+    {
+      "name" : "Clinical Quality Information WG",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:20]:
```suggestion
      "name" : "FHIR Infrastructure WG",
```

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+      "telecom": [
+        {
+          "system": "url",
+          "value" : "http://www.hl7.org/Special/committees/cqi"
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:24]:
```suggestion
          "value" : "http://www.hl7.org/Special/committees/fiwg"
```

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+      },
+      {
+        "id": "Extension.url",
+        "path": "Extension.url",
+        "type": [
+          {
+            "code": "uri"
+          }
+        ],
+        "fixedUri": "http://hl7.org/fhir/StructureDefinition/cqf-event-recorded"
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:65]:
```suggestion
        "fixedUri": "http://hl7.org/fhir/StructureDefinition/event-recorded"
```

**Review by brynrhodes — CHANGES_REQUESTED** [review]

**Comment by brynrhodes** [comment]
Also, please change the name of the file to match the new name of the extension (i.e. drop the cqf-)

File: input/definitions/Resource/StructureDefinition-cqf-event-recorded.json (hunk 6 above)
Comment (bryantaustin13) [reply input/definitions/Resource/StructureDefinition-cqf-event-recorded.json:6]: fhir-i causes an error as it is not in the valueset (http://hl7.org/fhir/ValueSet/hl7-work-group|5.1.0)
 The following is there and used instead.
fhir | FHIR Infrastructure | FHIR Infrastructure (http://www.hl7.org/Special/committees/fiwg/index.cfm).

**Review by brynrhodes — APPROVED** [review]

**Review by grahamegrieve — COMMENTED** [review]
Communication already has sent and received date. what's the point of this?

**Comment by brynrhodes** [comment]
@grahamegrieve , it supports the recorded date when a communication was not done, on review with FHIR-I there was agreement that sent and received didn't quite capture that notion, so event-recorded was suggested as an extension to support it.

**Comment by grahamegrieve** [comment]
Then the definition needs to be clear about what it is for, and what it is not. I didn't divine that from the existing definition. And you have.... about 6 hours.... to make it into the release

File: input/definitions/Resource/StructureDefinition-event-recorded.json [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+    {
+      "name" : "FHIR Infrastructure WG",
+      "telecom": [
+        {
+          "system": "url",
+          "value" : "http://www.hl7.org/Special/committees/fiwg"
+        }
+      ]
+    }
+  ],  "description": "Captures the recorded date of the event.",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-event-recorded.json:28]:
```suggestion
  ],  "description": "Captures the recorded date of the event.",
  "purpose": "This extension is defined to allow resources that do not have a suitable recorded date element to capture the time the event is recorded. For example, Communication resources in not-done status do not have an element to capture when the event was recorded.",
```

**Review by brynrhodes — CHANGES_REQUESTED** [review]

**Comment by brynrhodes** [comment]
Okay, added a purpose element to note the expected usage, does that help?

**Comment by grahamegrieve** [comment]
I'm approving this so it makes it into the ballot, but I don't see how that solves the problem I have. if you get this extension, what kind of event date is it? The definition actually muddies the water greatly. I don't see how it serves interoperability.

**Review by grahamegrieve — APPROVED** [review]
ok

**Comment by brynrhodes** [comment]
Thank you for merging, I've added this tracker to make sure we address the concern: https://jira.hl7.org/browse/FHIR-46490

**Pull request merged**